tmc-langs-java/src/main/java/fi/helsinki/cs/tmc/langs/java/maven/MavenInvokatorMavenTaskRunner.java

```diff
@@ -75,4 +93,54 @@ public void consumeLine(String line) {
             throw new MavenExecutorException(e);
         }
     }
+
+    private Path useBundledMaven() {
+        Path mavenHome = getConfigDirectory();
+        if (Files.exists(mavenHome)) {
+            log.info("Maven already extracted");
+
+            // Add the name of the extracted folder to the path
+            return mavenHome.resolve("apache-maven-3.3.9");
+        }
+        log.info("Maven bundle not previously extracted, extracting...");
+        try {
+
+            InputStream data = getClass().getResourceAsStream("apache-maven-3.3.9.zip");
+            Preconditions.checkNotNull(data,
+                "Couldn't load bundled maven from tmc-langs-java.jar.");
+            Path tmpFile = File.createTempFile("tmc-maven", "zip").toPath();
```

Comment:
No reason not to use `Files.createTempFile(...)` instead: return `Path` and has better permissions.